incubator/dokuwiki/templates/secrets.yaml

```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ template "fullname" . }}
+  labels:
+    app: {{ template "fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+type: Opaque
+data:
+  {{- if .Values.dokuwikiPassword }}
+  dokuwiki-password: {{ default "" .Values.dokuwikiPassword | b64enc | quote }}
```

Comment:
Since the `if` condition  already checks if `.Values.dokuwikiPassword` is not an empty string, this line could be changed to `{{ .Values.dokuwikiPassword | b64enc | quote }}`